site/content/tutorial/02-reactivity/04-updating-arrays-and-objects/text.md

```diff
@@ -31,11 +33,22 @@ function addNumber() {
 }
 ```
 
-A simple rule of thumb: the name of the updated variable must appear on the left hand side of the assignment. For example this...
+However, indirect assignments to references such as this...
```

Comment:
This sort of seems under-specified to me because it works if done directly in a script:

https://svelte.dev/repl/cd63c5870fbb43a19b860f2d4bc2b126?version=3.47.0

I believe it has to be within a function not to work:

https://svelte.dev/repl/d03cee2e12f74823a3ab916161020038?version=3.47.0

Reply:
The first repl seems to work because the assignment happens before the initial render. An example with the effect observable would need a lot more code and explanation, whereas I think the current one gets the point across succinctly.

Reply:
Would it really require a lot more code? All the examples above this are inside an `function addNumber() { ... }`. Wouldn't it make sense to also put these last two examples inside something like `function doSomething() { ... }` for consistency?

Reply:
The `addNumber` snippets refer to existing code on the right.

Having an example with the effects visible for the latter snippets would be [this much code](https://svelte.dev/repl/a4a92b55eb0c417da40af7012decfe1f?version=3.47.0) give or take, at which point it warrants its own repl/chapter.

Anything in between a full example and the current state doesn't add much imo.

Reply:
And we're saying "won't trigger reactivity" not that `obj` won't change itself which still holds.